File: claude/.claude/hooks/danger-guard.ts
import { readInput, notify } from "./shared";

interface PreToolUseInput {
  session_id: string;
  hook_event_name: "PreToolUse";
  tool_name: string;
  tool_input: {
    command?: string;
    [key: string]: unknown;
  };
  cwd: string;
}

const DANGEROUS_PATTERNS: { pattern: RegExp; label: string }[] = [
  { pattern: /rm\s+-[rf]{1,2}\b/i,          label: "rm -rf" },
  { pattern: /git\s+reset\s+--hard/i,        label: "git reset --hard" },
  { pattern: /git\s+push\s+(-f|--force)/i,   label: "git push --force" },
  { pattern: /git\s+clean\s+-[a-z]*f/i,      label: "git clean -f" },
  { pattern: /git\s+checkout\s+--\s/i,       label: "git checkout --" },
  { pattern: /git\s+branch\s+-D\s/i,         label: "git branch -D" },
  { pattern: /DROP\s+TABLE/i,                label: "DROP TABLE" },
  { pattern: /DROP\s+DATABASE/i,             label: "DROP DATABASE" },
  { pattern: /TRUNCATE\s+TABLE/i,            label: "TRUNCATE TABLE" },
  { pattern: /chmod\s+-R\s+777/i,            label: "chmod -R 777" },
  { pattern: /:\s*>\s*\S/,                   label: "> file truncate" },
];

const input = readInput<PreToolUseInput>();
const command = input.tool_input?.command ?? "";

const matches = DANGEROUS_PATTERNS.filter((p) => p.pattern.test(command));
if (matches.length > 0) {
  const labels = matches.map((m) => m.label).join(", ");
  const preview = command.slice(0, 80).replace(/\n/g, " ");
  notify("!! Danger Guard !!", `${labels}: ${preview}`, "notification");
}
// Always exit 0 — notify only, never block

File: codex/.codex/hooks/danger_guard.py
#!/usr/bin/env python3
"""Warn about destructive shell commands before Codex executes them."""

import re

from common import first, notify, read_event, tool_input

event = read_event()
data = tool_input(event)
command = str(first(data, "command", "cmd", default=""))
patterns = [
    (r"rm\s+-[rf]{1,2}\b", "rm -rf"),
    (r"git\s+reset\s+--hard", "git reset --hard"),
    (r"git\s+push\s+(-f|--force)", "git push --force"),
    (r"git\s+clean\s+-[a-z]*f", "git clean -f"),
    (r"git\s+checkout\s+--\s", "git checkout --"),
    (r"git\s+branch\s+-D\s", "git branch -D"),
    (r"DROP\s+(TABLE|DATABASE)", "DROP database object"),
    (r"TRUNCATE\s+TABLE", "TRUNCATE TABLE"),
    (r"chmod\s+-R\s+777", "chmod -R 777"),
    (r":\s*>\s*\S", "file truncate"),
]
matches = [label for pattern, label in patterns if re.search(pattern, command, re.IGNORECASE)]
if matches:
    notify("Codex danger guard", f"{', '.join(matches)}: {command[:100]}", "notification")
